idseq_dag/util/m8.py

```diff
@@ -630,3 +665,13 @@ def should_keep(hits):
         return is_whitelisted(hits) and not is_blacklisted(hits)
 
     return should_keep
+
+
+def _safe_sync(db, line_count=0):
+    try:
+        db.sync()
+    except Exception as e:
+        log.write("Error in sync of temp db: {}. Trying once more.".format(e), warning=True)
+        db.sync()
```

Comment:
This seems hacky. Why would we indiscriminately ignore the first error and expect the second sync to succeed?

Reply:
Yes it is hacky. I added it so I could test a run again. See https://user-images.githubusercontent.com/28797/78467260-87f47f00-76bf-11ea-87e6-b5873a5c41e0.png

Reply:
Seems to have helped partially in one run. 

![image](https://user-images.githubusercontent.com/28797/78503674-f9aee600-771c-11ea-8db6-eb492a632018.png)

Reply:
Without knowing what the original error was, I'm not comfortable using this approach. We should at the very minimum print the error type and message here.

Reply:
The exception we can now see above is an instance of `dbm.error` with the uninformative message `cannot add item to database`. While the retry is helping, these errors internal to `shelve`, unrelated to the original OOM error, have caused me to doubt the value of the whole PR. I don't think we can tell whether it would do more harm than good at this point without a lot more testing, and I'm not sure we want to invest that effort.